test/e2e/example_k8petstore.go

```diff
@@ -118,30 +118,31 @@ T:
 	for {
 		select {
 		case <-transactionsCompleteTimeout:
-			Logf("Timeout %v reached, transactions not complete.  Breaking!", tick)
+			Logf("PetStore : Completion timeout %v reached, transactions not complete.  Breaking!", tick)
 			break T
 		case <-startupTimeout:
-			err, totalTransactions = readTransactions(c, ns)
-			Logf("Timeout %v reached.  Checking if transactions have occured.", startupTimeout)
-			// If we don't have 3 transactions, fail the test.
-			if err != nil {
-				Logf("Failed : Error %v", err)
-				break T
-			}
-			if totalTransactions < 3 {
+			if !ready {
+				Logf("PetStore : Startup Timeout %v reached: Its been too long and we still haven't started accumulating transactions!", startupTimeout)
 				break T
 			}
-			ready = true
 		case <-tick:
-			// Pass if we've collected enough transactions
+			// Don't fail if there's an error.  We expect a few failures might happen in the cloud.
 			err, totalTransactions = readTransactions(c, ns)
-			if ready {
-				Expect(err).NotTo(HaveOccurred())
-			}
-			if totalTransactions > minExpected {
-				break T
+			if err == nil {
+				Logf("PetStore : Time: %v, %v = total petstore transactions stored into redis.", time.Now(), totalTransactions)
+				if totalTransactions >= 3 {
+					ready = true
+				}
+				if totalTransactions > minExpected {
```

Comment:
It probably doesn't matter, since we're expecting transactions to happen in large bursts... But yeah I guess the test is more precisely defined as >=